clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -715,6 +715,12 @@ public void abortTransaction() throws ProducerFencedException {
      * producer and create a new instance.
      * </p>
      * <p>
+     * If the message format of the destination topic is not upgraded to 0.11.0.0, produce requests will
```

Comment:
Sure I can clarify. The awkwardness of adding `@throws` lines is that `send` itself does not throw the exception: it is raised in the callback.